src/components/TodoList.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useContext } from 'react';
+import { TodoItem } from './TodoItem';
+import { TodosContext } from './TodosContext';
+import * as todoService from '../api/todos';
+import { Todo } from '../types/Todo';
+import { FilterOption } from '../types/FilterOption';
+
+export const TodoList: React.FC = () => {
+  const { todos, filterOption, setErrorMessage } = useContext(TodosContext);
+
+  const visibleTodos = todos.filter(todo => {
+    switch (filterOption) {
+      case FilterOption.Active:
+        return !todo.completed;
+      case FilterOption.Completed:
+        return todo.completed;
+      case FilterOption.All:
+      default:
+        return true;
+    }
+  });
+
+  const deleteTodo = (id: number) => {
+    return todoService.deleteTodo(id)
+      .catch(() => {
+        setErrorMessage('Unable to delete a todo');
+      });
+  };
+
+  const updatedTodo = (newTodo: Todo) => {
+    return todoService.updatedTodo(newTodo)
+      .catch((error) => {
+        setErrorMessage('Unable to update a todo');
+        throw error;
+      });
+  };
```

Comment:
Hey! I see you've fixed such issues exactly in the places I marked

But when you see some comments you should fix all similar cases:
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/86923089/9d730be3-6af7-4d58-ae35-31a98a47f2d5)
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/86923089/be309b71-05e0-49a4-822b-da54eaad6b3e)